packages/edit-site/src/components/left-sidebar/navigation-panel/index.js

```diff
@@ -32,39 +33,49 @@ const NavigationPanel = () => {
 		}
 	}, [ ref ] );
 
-	const { templateId, templatePartId, templateType, activeMenu } = useSelect(
-		( select ) => {
-			const {
-				getTemplateId,
-				getTemplatePartId,
-				getTemplateType,
-				getNavigationPanelActiveMenu,
-			} = select( 'core/edit-site' );
-
-			return {
-				templateId: getTemplateId(),
-				templatePartId: getTemplatePartId(),
-				templateType: getTemplateType(),
-				activeMenu: getNavigationPanelActiveMenu(),
-			};
-		},
-		[]
-	);
+	const {
+		templateId,
+		templatePartId,
+		templateType,
+		activeMenu,
+		page,
+	} = useSelect( ( select ) => {
+		const {
+			getTemplateId,
+			getTemplatePartId,
+			getTemplateType,
+			getNavigationPanelActiveMenu,
+			getPage,
+		} = select( 'core/edit-site' );
+
+		return {
+			page: getPage(),
+			templateId: getTemplateId(),
+			templatePartId: getTemplatePartId(),
+			templateType: getTemplateType(),
+			activeMenu: getNavigationPanelActiveMenu(),
+		};
+	}, [] );
 
 	const {
 		setTemplate,
 		setTemplatePart,
 		setNavigationPanelActiveMenu,
+		setPage,
```

Comment:
I've just realized this, and I've implicitly mentioned it in another comment, but I kinda feel like we could move some of these actions closer to where they are used, because, I might be wrong, it doesn't look like they are needed here.

So, `setPage` could live directly in the content menu (`content.js`).

`setTemplate` and `setTemplatePart` could be moved in their corresponding menus (`template.js` and `template-part.js`), or even both into `template-navigation-items.js` (using `entityType` to determine which to use).

(Note: if we agree on this move, we can take care of template and template part elsewhere)

Reply:
This sounds good to me! Probably `template-navigation-items.js` is the best place for it. It will make the component pretty smart though 😬